tests/basic/gdb/Makefile

```diff
@@ -0,0 +1,41 @@
+include ../../common.mk
+
+PROG=helloworld
+PROG_SRC=$(PROG).c
+IMAGE_SIZE=5M
+
+EXECUTION_TIMEOUT=60
+
+SGXLKL_ENV=SGXLKL_VERBOSE=1 SGXLKL_KERNEL_VERBOSE=1
+SGXLKL_HW_PARAMS=--hw-debug
+SGXLKL_SW_PARAMS=--sw-debug
+
+SGXLKL_ROOTFS=sgx-lkl-rootfs.img
+
+.DELETE_ON_ERROR:
+.PHONY: all clean
+
+$(SGXLKL_ROOTFS): $(PROG_SRC)
+	${SGXLKL_DISK_TOOL} create --size=${IMAGE_SIZE} --docker=./Dockerfile ${SGXLKL_ROOTFS}
+
+gettimeout:
+	@echo ${EXECUTION_TIMEOUT}
+
+run: run-hw run-sw
+
+run-gdb: run-hw-gdb
+
+run-hw: ${SGXLKL_ROOTFS}
+	  $(SGXLKL_ENV) $(SGXLKL_GDB) -batch -ex "handle SIGILL nostop noprint" -ex run -ex bt --args $(SGXLKL_STARTER) $(SGXLKL_HW_PARAMS) $(SGXLKL_ROOTFS) $(PROG)
+
+run-sw: ${SGXLKL_ROOTFS}
+	  $(SGXLKL_ENV) $(SGXLKL_GDB) -batch -ex "handle SIGILL nostop noprint" -ex run -ex bt --args $(SGXLKL_STARTER) $(SGXLKL_SW_PARAMS) $(SGXLKL_ROOTFS) $(PROG)
```

Comment:
Can you document what this test is supposed to do? How do you distinguish success vs failure?

Reply:
I did some more tests and it turns out due to a [GDB bug](https://sourceware.org/bugzilla/show_bug.cgi?id=13000) fixed in 8.3 (available in Ubuntu >= 19.10) gdb will always exit with 0 in batch mode. This means this test will never fail.

I guess to make the test generally more useful (and which would work around the issue above) I could run gdb commands that simulate a normal developer, like setting a breakpoint in an in-enclave function, get a backtrace, dump a value, etc. and compare that to expected output. We'd have to find something that is as stable as possible and doesn't depend on compiler optimizations etc. otherwise the test would break eventually.

Reply:
I modified the test to set a breakpoint and check that it gets hit including the right function argument value. Can you check whether this is self-explanatory or it needs a comment?

Reply:
I think that's clear.